modules/sap_hana/main.tf

```diff
@@ -97,9 +99,13 @@ resource "google_compute_instance" "gcp_sap_hana" {
     subnetwork         = var.subnetwork
     subnetwork_project = var.project_id
 
-    access_config {
-      nat_ip = google_compute_address.gcp_sap_hana_ip.address
+    dynamic "access_config" {
+      for_each = var.public_ip == 1 ? google_compute_address.gcp_sap_hana_ip : []
+      content {
+        nat_ip = access_config.value.address
```

Comment:
I don't think this is a valid value. What does it refer to?

Reply:
google_compute_address.gcp_sap_hana_ip refers to the external IP that gets created above. I added a count to that (above) and that makes it an array. That is why it works quite well with the for_each here.

So, yes I tested it and it is working properly.

Reply:
Got it, I was having trouble following but that does indeed look quite sensible.